src/Lib/Managers/SelectedBoundManager.vala

```diff
@@ -66,6 +73,10 @@ public class Akira.Lib.Managers.SelectedBoundManager : Object {
         initial_event_x = event_x;
         initial_event_y = event_y;
 
+        initial_drag_press_x = event_x;
+        initial_drag_press_y = event_y;
+        initial_drag_item_registered = false;
```

Comment:
I think we need a comment here to explain why we're setting this variable to false when a new event is triggered.